### ISSUE TITLE
Should manpages link to non-existant docs?

### ISSUE THREAD

**Issue by richsalz** [openssl/openssl#10109]
I recently created a half-dozen PR's to fix broken links: #10104 #10103 #10102 #10101 #10100 #10097.  Most of the changes in those PR's are to remove links (L<>) to pages that don't exist. In almost all cases, I turned it into just an API link.

@levitte believes that the links should still exist, to add incentive to write the docs; one of those get a thumbs-up emoji from @t8m.  I strongly disagree and feel that we should not knowingly publish manpages that have broken links. Part of the reason is that Richard found several function names that were mis-spelled; it is unlikely we would have ever found them had I not removed all broken links.  Another reason is that we have another tool (the -d/-e/-u/-v flags to find-doc-nits) to list undocumented API's.  If we want to prioritize for functions that are documented in the existing manpages, we can enhance the tool to do that.

What does the project want to do?

**Comment by beldmit** [comment]
IMHO if there is a link to an existing undocumented public API function, it's worth to add the link and maybe create a stub page. Some of these stub pages should be converted to GH issues and labeled as good first issues :)

BTW, from time to time I meet functions that are just enlisted in the header of the man page but not documented at all really, and it's necessary to deal with it.

**Comment by levitte** [comment]
The issue I see with converting `L<foo(n)>` for a yet undocumented `foo` is not only does it make it less directly visible (and the missing docs feature will only catch undocumented *functions*, not yet-to-be-written pages like `doc/man7/openssl-core-numbers.pod`), but it also adds more work on those authoring those missing pages, 'cause they will have chase down every spot where there should be a link.

**Comment by levitte** [comment]
Furthermore, you're trying to make these changes in *master*, which is under heavy development and in a certain state of flux. Removing the links in such a branch is extra exasperating.

**Comment by mspncp** [comment]
> one of those get a thumbs-up emoji from @t8m. 

If you are referring to https://github.com/openssl/openssl/pull/10103#pullrequestreview-297824200, then it was my thumb. I agree with @levitte's opinion https://github.com/openssl/openssl/issues/10109#issuecomment-538758448 that removing those cross references (even if the target does not exist yet) removes valuable information which is not easily restored.

**Comment by richsalz** [comment]
Yes, I am trying to make them in master.  I thought it wouldn't get approved in a release branch.  They backport pretty easily.

As for optimizing the broken links so that documents can be written, and the referring documents updated, this works:
```
# Find undocumented functions
./util/find-doc-nits -d \
    | grep function \
    | awk '{printf "%s()\n",  $3;}' > /tmp/list
# Find inline references to them
cd doc
git grep -f /tmp/list
```

I think this little script addresses some concerns.

**Comment by levitte** [comment]
What about things that are not functions (i.e potentially everything that isn't man3)?

**Comment by richsalz** [comment]
Can you give me an example?  Perhaps the "find undocumented" flags of find-doc-nits need enhancement.

**Comment by levitte** [comment]
https://github.com/openssl/openssl/issues/10109#issuecomment-538758448

**Comment by richsalz** [comment]
Thank you, you are right, that linked comment points out something other than a function.

I still think the fixes are worth doin and you don't.  Disagreement is okay.  Would it be good enough if I opened one or more issues to capture the non-function changes?

**Comment by levitte** [comment]
No.

Look, *master* is in development so incredibly inappropriate for this kind of strictness.  However, when *3.0* is starting to close on a final release, *then* this kind of work is appropriate, *on that release branch*.

Also, nothing stops anyone from raising issues about specific missing documentation, such as "Missing documentation for blah()", or "Missing document doc/man7/openssl-core-numbers.h".  That would allow anyone from the community to pick up on it and help out.  I would say that's much more appropriate than to remove links early.

**Comment by richsalz** [comment]
Thank you for explaining.  We disagree.  I don't see it as strictness, I see it as broken docs, which should be no more acceptable than broken code.  But I acknowledge that others disagree.

I haven't seen anyone else on the project support my view, so I guess the PR's can be closed.  Feel free to rummage through them and see if there are other fixes folks want.

**Comment by levitte** [comment]
Did you notice that there were things in those PRs that no one had any issue with? How about cleaning so they can be merged? Or are you giving us an all-or-nothing deal?

**Comment by t-j-h** [comment]
Broken docs are **nothing** like broken code - we don't run the docs - and in fact installing the documentation isn't something you have to do. 
This is open source code - the source itself is available. 
Suggesting that docs and code are equivalent in terms of priority is not something that I think is supportable.
The docs that are broken should be fixed - but docs are definitely not at the same priority as the code as they are fundamentally different things.

**Comment by richsalz** [comment]
If it were all-or-nothing I would close the PR's and delete the branches.  Instead, the offer is "okay, you don't want this, pick out what you do want."  I do not have the mental resources (nor the heart, truth be told) to make even educated guesses as to what the project wants.

It's a smorgesbord, pick and choose as you wish.  Don't ask me to fill your plate. :)

**Comment by richsalz** [comment]
To summarize what's been happening over in PR-land.  There are a handful of PR's to address and fix broken links (#10104 and #10100 are the remaining ones), and a "final" #10256 that would address the "missing link" situation overall and includes a "Fixes" line for this issue.
